panels_data.py

```diff
@@ -0,0 +1,143 @@
+from sqlalchemy import create_engine, Column, Integer, String, Float
+from sqlalchemy.orm import declarative_base, sessionmaker
+from sqlalchemy.engine import URL
+
+Base = declarative_base()
+
+
+# creating Panel class to declare the structure of the database table and define what the entries look like
+class Panel(Base):
+    __tablename__ = "panels_data"
+
+    id = Column(Integer(), primary_key=True, autoincrement=True)
+    name = Column(String(100), nullable=False, unique=True)
+    stack = Column(Integer(), nullable=False)
+    efficiency = Column(Float(), nullable=False)
+    num_panels = Column(Integer(), nullable=False)
+    tilt = Column(Float(), nullable=False)
+
+
+def create_panel_table(db_user, db_password, db_host, db_name):
+    # connecting to database
+    engine = create_engine(
+        f"postgresql+psycopg2://{db_user}:{db_password}@{db_host}/{db_name}"
+    )
+
+    # creates corresponding tables
+    Base.metadata.create_all(engine)
```

Comment:
I think this will fail if the table already exists. Based on your previous code, you should try to use that logic. I think this can be done like this
```suggestion
    Base.metadata.create_all(engine, checkfirst=True)
```
But I might be wrong, so please confirm this and use accordingly.

Reply:
It didn't work but I figure out how to drop the table if it exists first before creating the table and tested it. Should be fine now.